### PR TITLE
Add  new Jenkinsfile with versioning

### PR DESCRIPTION
**Pull request by MaksymukNatalia**
## Task

* [Ticket](https://github.com/orgs/ita-social-projects/projects/21/views/10?sliceBy%5Bvalue%5D=MaksymukNatalia&pane=issue&itemId=45936766)

## Summary 

Remake build steps to Jenkinsfile. Now Jenkinsfile provide completed CI/CD.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by IvannYad — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,53 +1,288 @@
...
+                sh 'export DOCKER_TAG_FRONTEND=${preDeployFrontProd}'
+                sh 'docker compose down && sleep 10'
+                sh 'docker compose --env-file /etc/environment up -d'
+                  
+            }
+        }
+        post {
+            always {
+                echo 'Always'
+            }
```
Comment (LanchevychMaxym) [inline Jenkinsfile:214]: It looks like it's not used

File: Jenkinsfile [hunk 2]
```diff
@@ -1,53 +1,288 @@
...
+                sh 'docker system prune --force --filter "until=72h"'
+                sh 'export DOCKER_TAG_BACKEND=${preDeployBackStage}'
+                sh 'export DOCKER_TAG_FRONTEND=${preDeployFrontStage}'
+                sh 'docker compose down && sleep 10'
+                sh 'docker compose --env-file /etc/environment up -d'
+                
+
+            }
+        }
+    }
```
Comment (LanchevychMaxym) [inline Jenkinsfile:240]: We already have this on lines 176-184, right?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,53 +1,252 @@
...
+                      sh 'docker system prune --force --filter "until=72h"'
+                      sh 'export DOCKER_TAG_BACKEND=${env.CODE_VERSION}'
+                      sh 'export DOCKER_TAG_FRONTEND=${preDeployFrontProd}'
+                      sh 'docker compose down && sleep 10'
+                      sh 'docker compose --env-file /etc/environment up -d'
+                }
+        }
+        post {
+          aborted{
+              input message: 'Do you want to rollback deploy stage?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak , dev'
```
Comment (LanchevychMaxym) [inline Jenkinsfile:181]:
```suggestion
              input message: 'Do you want to rollback Staging deployment?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak , dev'
```

File: Jenkinsfile [hunk 4]
```diff
@@ -1,53 +1,252 @@
...
+                sh "gh release create v${vers}  --generate-notes --draft"
+            }
+        }
     }
+    stage('Rollback Prod') {  
+        when {
+           expression { isSuccess == '1' }
+            }
+        steps {
+            input message: 'Do you want to rollback deploy prod?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak '
```
Comment (LanchevychMaxym) [inline Jenkinsfile:232]:
```suggestion
            input message: 'Do you want to rollback Production deployment?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak '
```

File: Jenkinsfile [hunk 5]
```diff
@@ -1,53 +1,252 @@
...
+                    sh "docker system prune --force --filter "until=72h"'
+                    sh 'export DOCKER_TAG_BACKEND=${env.CODE_VERSION}'
+                    sh 'export DOCKER_TAG_FRONTEND=${preDeployFrontStage}'
+                    sh 'docker compose down && sleep 10'
+                    sh 'docker compose --env-file /etc/environment up -d'
+
+                }  
+            }
+     }    
+    stage('WHAT IS THE NEXT STEP') {
```
Comment (LanchevychMaxym) [inline Jenkinsfile:160]:
```suggestion
    stage('Deploy Production') {
```

File: Jenkinsfile [hunk 6]
```diff
@@ -1,53 +1,252 @@
...
+       when {
+                expression { IS_IMAGE_PUSH == true }
+            }  
+        steps {
+            input message: 'Do you want to deployProd deploy prod?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak'
```
[the remark this answers is not in the document]
Comment (LanchevychMaxym) [reply Jenkinsfile:165]: If I'm not wrong here should be added Prod agent label

**Review by yaakymiv — APPROVED** [review]

**Review by Lazy-Lenny — APPROVED** [review]

File: Jenkinsfile [hunk 7]
```diff
@@ -1,53 +1,275 @@
...
                     }
                 }
             }
         }
+    stage('Deploy Stage'){
+        when {
+                expression { IS_IMAGE_PUSH == true }
+            }  
+        steps {
+            input message: 'Do you want to approve deploy stage?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak , dev'
```
Comment (LanchevychMaxym) [inline Jenkinsfile:139]:
```suggestion
            input message: 'Do you want to approve Staging deployment?', ok: 'Yes', submitter: 'admin_1, ira_zavushchak , dev'
```

**Review by LanchevychMaxym — APPROVED** [review]

**Pull request merged**